auth.go

```diff
@@ -438,43 +452,51 @@ func (x *Central) Login(username, password string) (sessionkey string, token *To
 		return sessionkey, token, err
 	}
 
-	x.Log.Infof("Login authentication success (%v)", userId)
+	x.Log.Infof("Login authentication success (%v)", user.UserId)
 
 	var permit *Permit
-	if permit, err = x.permitDB.GetPermit(userId); err != nil {
-		x.Log.Infof("Login GetPermit failed (%v) (%v)", userId, err)
+	if permit, err = x.permitDB.GetPermit(user.UserId); err != nil {
+		x.Log.Infof("Login GetPermit failed (%v) (%v)", user.UserId, err)
 		return sessionkey, token, err
 	}
 
 	if x.MaxActiveSessions != 0 {
-		if err = x.sessionDB.InvalidateSessionsForIdentity(userId); err != nil {
-			x.Log.Warnf("Invalidate sessions for identity (%v) failed when enforcing MaxActiveSessions (%v)", userId, err)
+		if err = x.sessionDB.InvalidateSessionsForIdentity(user.UserId); err != nil {
+			x.Log.Warnf("Invalidate sessions for identity (%v) failed when enforcing MaxActiveSessions (%v)", user.UserId, err)
 			return sessionkey, token, err
 		}
 	}
 
-	token = &Token{}
-	token.Expires = time.Now().Add(x.NewSessionExpiresAfter)
-	token.Permit = *permit
-	token.Identity = identity
-	token.UserId = userId
+	token = &Token{
+		Permit:   *permit,
+		Identity: user.getIdentity(),
+		UserId:   user.UserId,
+	}
+
+	userPasswordExpiry := user.PasswordModifiedDate.Add(x.PasswordExpiresAfter)
+	sessionExpiry := time.Now().Add(x.NewSessionExpiresAfter)
+	if userPasswordExpiry.After(time.Now()) && userPasswordExpiry.Before(sessionExpiry) {
```

Comment:
I don't understand the role of `userPasswordExpiry.After(time.Now())`. As far as I can tell, it shouldn't be there.